SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1441 @@
+			$result.productAttributes | Add-Member -notepropertyname "driftMessage" -notepropertyvalue ""
+			$ppmEnvironmentalDetails.Add($result)
+			Write-LogMessage -type INFO -Message "Retrieving WSA server password policies..complete"
+		} else {
+			Write-LogMessage -Type WARNING -Message "Workspace One Access Skipped.  Login details are not provided."  
+		}
+
+		$ppmVariables.isEnvDetailSet = $true
+		Write-LogMessage -type INFO -Message "Retrieving Environment Details....Completed" -Colour Green
+	}
+}
+
+################################################################################
+####   Print current environment password policies settings as html file	####
+################################################################################
+Function ExportReportHTML {
+	Param (
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [psobject]$ppmVariables,
+		[Parameter (Mandatory = $true)] [AllowEmptyCollection()][System.Collections.Generic.List[System.Object]]$ppmStandardConfigValues,
+        [Parameter (Mandatory = $true)] [AllowEmptyCollection()][System.Collections.Generic.List[System.Object]]$ppmEnvironmentalDetails,
+		[Parameter (mandatory = $false)] [Switch]$doComparison = $false
+	) 
+		#retrieving environmental details
+	GetEnvironmentPasswordPolicyDetail -ppmVariables $ppmVariables -ppmEnvironmentalDetails $ppmEnvironmentalDetails
+	$count = 0
+		#addressing comparison reports
+	if ($doComparison -eq $true) {
+		foreach ($eachNode in $ppmEnvironmentalDetails) {
+			foreach ($eachAttribute in $eachNode.productAttributes.PSObject.Properties){
+				foreach ($product in $ppmStandardConfigValues) {
+					if ($product.productType -eq $eachNode.productType) {
+						if ($eachAttribute.Value -ne $null) {
+							if ($eachAttribute.Value -ne $product.productAttributes.$($eachAttribute.Name)) {
+								$eachAttribute.Value = "<font color=#DC143C><b>" + $eachAttribute.Value + "</b></font>["+$product.productAttributes.$($eachAttribute.Name)+"]"
+								$count += 1
+							}
+						} else{
+							$eachAttribute.Value = "<font color=#DC143C><b>NULL</b></font>["+$product.productAttributes.$($eachAttribute.Name)+"]"
+							$count += 1
+						}
+					}
+				}
+			}
+		}
+	}
+	#prepare html header
+	$htmlOutput = "<html><title>Password Policy Report</title><style type=" + '"text/css"' + ">.myTable { border-collapse:collapse; }.myTable td, .myTable th {text-align:center;padding:5px;border:1px solid #000;}</style><head><H1>Password Policy Report for the Workload domain <font color=#6495ED>$($ppmVariables.sddcDomainName)</font> </H1></head><body><br><H4><p>-Report Ran on <font color=#6495ED>" + (Get-Date).tostring("dd-MM-yyyy-hh-mm-ss") + "</font> for the Workload domain <font color=#6495ED>$($ppmVariables.sddcDomainName)</font>"
+	if ($doComparison -eq $false) {
+		$htmlOutput = $htmlOutput + "<br><font color=#FFA500>-Common password policy file not provided, Comparison has not been evaluated</font></p></H4>"
```

Comment:
Please review this sentence structure.

Reply:
addressed in 99029d5ec8015e0c5c645e9abc7c3384e1b3091e